jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java

```diff
@@ -591,46 +647,151 @@ public String toString()
         FAILURE
     }
 
-    private class Decoder extends IteratingNestedCallback
+    private class ContentListeners
```

Comment:
Javadoc on this class saying what it is doing would be appreciated by our future selves.